core/src/main/scala/org/apache/spark/status/protobuf/Utils.scala

```diff
@@ -17,10 +17,24 @@
 
 package org.apache.spark.status.protobuf
 
+import com.google.protobuf.MessageOrBuilder
+
 object Utils {
   def getOptional[T](condition: Boolean, result: () => T): Option[T] = if (condition) {
     Some(result())
   } else {
     None
   }
+
+  def setStringField(input: String, f: String => MessageOrBuilder): Unit = {
```

Comment:
Should we make this function more general? Such as 

```
  def setField[T](input: T, f: T => MessageOrBuilder): Unit = {
    if (input != null) {
      f(input)
    }
  }
```
Other types of input may also be null, such as map.

https://github.com/apache/spark/blob/ec424c5b0e392acc57e825fb94a21d6963ebece9/sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/SQLExecutionUIDataSerializer.scala#L37-L41

Reply:
Sure

Reply:
On second thought, there is no convenient way for getting optional map. As per your suggestion, there will be `setField` and `getStringField`, which become asymmetrical. 
I will keep the current code

Reply:
Agree with you